app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,20 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def log_out
+    session.delete :user_id
+    @current_user = nil
+  end
+
+  def current_user
+    return unless session[:user_id]
+
+    @current_user ||= User.find_by id: session[:user_id]
```

Comment:
`||=` có tác dụng gì em

Reply:
Em xin trả lời câu hỏi như sau:
- Toán tử này được gọi là assignment operator, được sử dụng phổ biến trong ngôn ngữ lập trình. Về tổng quan nó có dạng như sau : expr1 op= expr2, tương đương với expr1 = (expr1) op (expr2),
- Trong đoạn mã trên nó tương đương với câu lệnh sau: '@current_user = @current_user || User.find_by id:session[:user_id]', trong ngữ cảnh này có nghĩa là nếu đã tồn tại user rồi thì gán user cũ, còn không thì time trong database để và sau đó gán.